tests/bootstrap_tests/test_backups.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+from __future__ import with_statement
+
+import itertools
+import os
+import sys
+from mock import patch
+import sgtk
+from sgtk.util import ShotgunPath
+from sgtk.bootstrap.configuration_writer import ConfigurationWriter
+from shutil import copytree, ignore_patterns
+
+from tank.descriptor import Descriptor, descriptor_uri_to_dict, descriptor_dict_to_uri, create_descriptor, constants
+from tank_test.tank_test_base import setUpModule # noqa
+from tank_test.tank_test_base import temp_env_var
+from tank_test.tank_test_base import TankTestBase
+
+
+class TestBackups(TankTestBase):
+    def setUp(self):
+        super(TestBackups, self).setUp()
+
+        self._tmp_bundle_cache = os.path.join(self.tank_temp, "bundle_cache")
+        self._resolver = sgtk.bootstrap.resolver.ConfigurationResolver(
+            plugin_id="backup_tests",
+            bundle_cache_fallback_paths=[self._tmp_bundle_cache]
+        )
+
+    def test_cleanup(self):
+        """
+        Ensures that after a successful update the backup folder created by the
+        update process is properly deleted 
+        """
+        pathHead, pathTail = os.path.split(__file__)
+        core_path=os.path.join(pathHead,"..", "..")
+        temp_test_path=os.path.join(pathHead, "..", "fixtures", "bootstrap_tests", "test_update")
```

Comment:
@jfboismenu I tried doing something similar to what you suggest, but i got this failure:

test_cleanup (bootstrap_tests.test_backups.TestBackups) ... ERROR

======================================================================
ERROR: test_cleanup (bootstrap_tests.test_backups.TestBackups)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\projects\shotgun\manual\tk-core\tests\bootstrap_tests\test_backups.py
", line 70, in test_cleanup
    config.update_configuration()
  File "D:\projects\shotgun\manual\tk-core\python\tank\bootstrap\cached_configur
ation.py", line 255, in update_configuration
    self._config_writer.create_tank_command()
  File "D:\projects\shotgun\manual\tk-core\python\tank\util\filesystem.py", line
 68, in wrapper
    return func(*args, **kwargs)
  File "D:\projects\shotgun\manual\tk-core\python\tank\bootstrap\configuration_w
riter.py", line 255, in create_tank_command
    for file_name in os.listdir(root_binaries_folder):
WindowsError: [Error 3] The system cannot find the path specified: 'c:\\users\\g
omezll\\appdata\\local\\temp\\tankTemporaryTestData_cac24c81-4078-440c-926b-3bcc
791bf21a\\unit_test_mock_sg\\site.backup_tests\\cfg\\install\\core\\setup\\root_
binaries/*.*'

----------------------------------------------------------------------
Ran 1 test in 3.691s

FAILED (errors=1)

Haven't been able to figure this out. Perhaps tomorrow I could take a look at it with your aid?